lib/splicer.rb

```diff
@@ -79,18 +78,30 @@ def self.create_record_in_zone(record, zone)
 
   # @return [void]
   def self.update_record_in_zone(record, zone)
-    return false unless zone.is_a?(Splicer::Zone)
-    return false unless record.is_a?(Splicer::Records::Record)
+    return false if arguments_are_invalid(record, zone)
     providers.each do |provider|
       provider.update_record_in_zone(record, zone)
     end
   end
 
+  # Fetches list of associated records
+  # @return [] WE'LL FIGURE THIS OUT AS WE GO
+  def self.get_records_for(zone)
+    return false if zone_is_invalid(zone)
+
+    records = []
+    providers.each do |provider|
+      result = provider.get_records_for(zone)
+      next unless result
+      result.each { |record| records << record }
+    end
+    records
```

Comment:
Can be shortened to `providers.map { |provider| provider.get_records_for(zone)}.flatten` Make sure that instead of returning nil for `#get_records_for` you return an empty array. Or you can get around this by doing `providers.map { |provider| provider.get_records_for(zone) || [] }.flatten` instead